nft/ethereum_payment_obligation.go

```diff
@@ -114,17 +119,36 @@ func (s *ethereumPaymentObligation) MintNFT(documentID []byte, docType, registry
 		return nil, err
 	}
 
+	return requestData, nil
+
+}
+
+// MintNFT mints an NFT
+func (s *ethereumPaymentObligation) MintNFT(documentID []byte, docType, registryAddress, depositAddress string, proofFields []string) (<-chan *WatchTokenMinted, error) {
+
+	requestData, err := s.prepareMintRequest(documentID, docType, proofFields)
+
 	opts, err := s.ethClient.GetTxOpts(s.config.GetEthereumDefaultAccountName())
 	if err != nil {
 		return nil, err
 	}
 
-	watch, err := s.setupMintListener(requestData.TokenID)
+	watch, err := s.setupMintListener(requestData.TokenID, registryAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	var contract *EthereumPaymentObligationContract
+	if registryAddress == "" {
```

Comment:
I think we can avoid this if checks if we can update the registryAddress at the API level itself. What do you think ?